File: app/src/main/java/com/philsoft/metrotripper/app/drawer/SearchViewHolder.java
package com.philsoft.metrotripper.app.drawer;

import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.EditText;
import android.widget.ImageView;

import com.philsoft.metrotripper.R;
import com.philsoft.metrotripper.utils.ui.Ui;

public class SearchViewHolder extends RecyclerView.ViewHolder {
    public View root;
    public EditText entry;
    public ImageView searchButton;


    public SearchViewHolder(View root) {
        super(root);
        this.root = root;
        entry = Ui.findView(root, R.id.search_drawer_item_entry);
        searchButton = Ui.findView(root, R.id.search_drawer_item_button);
    }
}

File: app/src/main/java/com/philsoft/metrotripper/app/drawer/SearchViewHolder.kt
package com.philsoft.metrotripper.app.drawer

import android.support.v7.widget.RecyclerView
import android.view.View
import android.view.inputmethod.EditorInfo
import com.jakewharton.rxbinding2.view.RxView
import com.jakewharton.rxbinding2.widget.editorActionEvents
import kotlinx.android.extensions.LayoutContainer
import kotlinx.android.synthetic.main.search_drawer_item.*

class SearchViewHolder(override val containerView: View) : RecyclerView.ViewHolder(containerView), LayoutContainer {
    private val searchAction = entry.editorActionEvents()
            .filter { event ->
                event.actionId() == EditorInfo.IME_ACTION_GO
            }
    val searchEvent = RxView.clicks(searchButton)
            .mergeWith(searchAction)
            .map { entry.text.toString() }
}